merlin/dag/graph.py

```diff
@@ -123,7 +123,8 @@ def _validate_node_schemas(self, root_schema, nodes, strict_dtypes=False):
     @property
     def input_schema(self):
         # leaf_node input and output schemas are the same (aka selection)
-        return _combine_schemas(self.leaf_nodes)
+        # subgraphs can also be leaf nodes now, so input and output are different
+        return _combine_schemas(self.leaf_nodes, input_schemas=True)
```

Comment:
This seems like a good place to use two separate methods (`_combine_input_schemas` and `combine_output_schemas`) instead of a boolean flag